File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.java
class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        Set<Integer> set = new HashSet<>();

        Arrays.sort(nums2);

        for (int n : nums1) {
            int idx = Arrays.binarySearch(nums2, n);
            if(idx >= 0) {
                set.add(n);
            }
        }

        return set.stream().mapToInt(i->i).toArray();
    }
}

File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.py
class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        if not nums1 or not nums2:
            return []

        ans = set()

        nums2.sort()


        for n in nums1:
            idx = bisect.bisect_left(nums2, n)
            if idx < len(nums2) and nums2[idx] == n:
                ans.add(n)

        return list(ans)
        